Resources/AzureDevOps/azure-pipelines.yml

```diff
@@ -14,20 +12,66 @@ pr:
     include:
     - master
 
-steps:
+stages:
 
-# ARM template
-- template: AzureDevOpsTemplates/Build/dfc-arm-build.yml@dfc-devops
-  parameters:
-    ArmTemplateRoot: '$(System.DefaultWorkingDirectory)\Resources'
+- stage: Build
+  jobs:
+  - job: TestAndPublish
+    pool:
+      name: 'NCS - CI and CD'
+    variables:
+      - group: 'KeyVault - dfc-dev-shared-kv'
+      - group: dfc-shared-infrastructure-dev
     steps:
-- task: CopyFiles@2
-  displayName: 'Copy PSScripts Files to: $(Build.ArtifactStagingDirectory)'
-  inputs:
-    Contents: 'PSScripts/**/*.ps1'
-    TargetFolder: '$(Build.ArtifactStagingDirectory)'
-- task: PublishBuildArtifacts@1
-  displayName: Publish Artifact
-  inputs:
-     pathtoPublish: $(Build.ArtifactStagingDirectory)
-     artifactName: drop
\ No newline at end of file
+    # ARM template
+    - task: TotalALM.totalalm-tokenization.tokenization-build-task.Tokenization@2
+      displayName: 'Tokenization: Transform file parameters.json'
+      inputs:
+        SourcePath: '$(System.DefaultWorkingDirectory)\Resources'
+        TargetFileNames: test-parameters.json
+    - template: AzureDevOpsTemplates/Build/dfc-arm-build.yml@dfc-devops
+      parameters:
+        ArmTemplateRoot: '$(System.DefaultWorkingDirectory)\Resources'
+        steps:
+    - task: CopyFiles@2
+      displayName: 'Copy PSScripts Files to: $(Build.ArtifactStagingDirectory)'
+      inputs:
+        Contents: 'PSScripts/**/*.ps1'
+        TargetFolder: '$(Build.ArtifactStagingDirectory)'
+    - task: PublishPipelineArtifact@0
+      displayName: Publish Pipeline Artifact
+      inputs:
+        targetPath: $(Build.ArtifactStagingDirectory)
+        artifactName: Dfc.Shared
+    # Also publish build artifact for backwards compatibility
+    - task: PublishBuildArtifacts@1
+      displayName: Publish Build Artifact
+      inputs:
+        pathtoPublish: $(Build.ArtifactStagingDirectory)
+        artifactName: drop
+
+- stage: DeployToDev
+  variables:
+  - group: dfc-shared-all
+  - group: dfc-shared-infrastructure-all
+  - template: VariableTemplates\ReleaseVariables.yml

```

Comment:
I don't think this should be here!

Reply:
Ignore that, the variable templates have different variables in them.
Maybe we need to look at their filename as not to cause confusion?

Reply:
Renamed files